doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -9,77 +9,75 @@ sort_idx: 10
-The resolution response from the Bootstrap service contains a prioritized list of Operations services with their connectivity details and 
-is signed with the Bootstrap service private key. 
-The endpoint verifies the signature of the response by using the Bootstrap public key available from the build time parameters and retains the received information.
+For security and load-balancing reasons, Kaa endpoint SDKs do not include any information about Operations services.
+Instead, a [Control service]({{root_url}}Glossary/#control-service) embeds a list of available Bootstrap services into the SDK during the SDK generation process.
+This list is saved as a properties file in case with Java SDK, and as a header file in case with other [SDK types]({{root_url}}Glossary/#sdk-type).
+The endpoint selects a random Bootstrap service from the list and sends a resolution request to the selected Bootstrap service.
+The resolution request contains the [application token]({{root_url}}Glossary/#application-token) that can be used by the Bootstrap service in specific load-balancing strategies.
+The resolution response from the Bootstrap service contains a prioritized list of Operations services with their connectivity details and is signed with the Bootstrap service private key.
+The endpoint verifies the signature of the response using the Bootstrap public key available from the build time parameters and saves the received information.
 
-### Endpoint Authorization
+### Endpoint authorization
 
-Endpoint authorization is done by validating endpoint credentials using corresponding credentials service. 
-Credentials service may be configured separately for each application use either Admin UI or 
-[Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Application/editApplication). 
+Endpoint authorization is done by validating the endpoint credentials using the corresponding credentials service.
+Credentials service can be configured for each Kaa client individually.
+To do this, you can use the [Administration UI]({{root_url}}Glossary/#administration-ui) or the corresponding [server REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Application/editApplication).
 
 ![credential](attach/credentials-for-apps.png)
 
-At the moment of writing Kaa endpoint credentials is identifier is based on hash of endpoint RSA Key Pair.
-During the start of new session, operations service lookup credentials service based on application identifier 
-and uses this credentials service to lookup status of provided credentials object. 
-If Credentials object is not found or this credentials are in use by other endpoint, new session is rejected by Kaa node.
-
-### Endpoint Registration information
+The endpoint credentials ID is generated based on the RSA key pair hash of the endpoint.
+When a new session starts, the Operations service challenges the Credentials service to retrieve credentials status.
```

Comment:
"challenges" is a special term and should not be used here to avoid confusion. Rather, use "queries", etc.